### PR TITLE
Docs home page and navigation refactor

### PR DESCRIPTION
**Pull request by discdiver**
To improve the docs experience for new and existing users we are proposing to add

1. A streamlined home page
1. Simplified navigation
1. Formatting enhancements
1. Changes "collections" to "integrations" for consistency and easier understanding
1. Other minor fixes

[Docs preview](https://deploy-preview-8807--prefect-docs.netlify.app/)

### Checklist

- [ ] This pull request references any related issue by including "closes `<link to issue>`"
	- If no issue exists and your change is not a small fix, please [create an issue](https://github.com/PrefectHQ/prefect/issues/new/choose) first.
- [x] This pull request includes tests or only affects documentation.
- [x] This pull request includes a label categorizing the change e.g. `fix`, `feature`, `enhancement`

### CODE REVIEW

File: mkdocs.yml [hunk 1]
```diff
@@ -163,14 +155,22 @@ nav:
         - Audit Log: ui/audit-log.md
         - Troubleshooting: ui/troubleshooting.md
         - Prefect Cloud API Rate Limits: ui/rate-limits.md
+    - Guides: 
+        - guides/index.md
+        - Testing: guides/testing.md
+        - Dask & Ray: guides/dask-ray-task-runners.md
+        - Deployment:
+            - Docker: guides/deployment/docker.md
+            - Azure Container Instance: guides/deployment/cloud-agent.md
```
Comment (EmilRex) [inline mkdocs.yml:164]: I messed up the link:
```suggestion
            - Azure Container Instance: guides/deployment/aci.md
```

File: mkdocs.yml [hunk 2]
```diff
@@ -93,38 +96,27 @@ plugins:
...
-                - tutorials/testing.md
-                - tutorials/dask-ray-task-runners.md
-        - Migration Guide: migration-guide.md
-        - Prefect Recipes: 
-            - Recipes Catalog: recipes/recipes.md
-            - Contributing Recipes: recipes/contribute-recipes.md
-    - Concepts:
-        - Overview: concepts/overview.md
+        - Cloud Quickstart: ui/cloud-quickstart.md
+    - Tutorials:
```
Comment (EmilRex) [inline mkdocs.yml:110]:
```suggestion
    - Tutorial: 
```

Comment (discdiver) [reply mkdocs.yml:110]: Right now it's still a bunch of tutorials, right? So do you think we should leave it as the plural until we condense?

Comment (EmilRex) [reply mkdocs.yml:110]: Well I guess it's semantic, but to me it's one tutorial... _the tutorial_ on how to use Prefect.

File: README.md [hunk 3]
```diff
@@ -18,9 +18,9 @@
...
 
-Prefect 2 is an orchestrator for data-intensive workflows. Powered by an asynchronous orchestration engine, its the easiest way to transform any Python function into a unit of work that can be observed and governed. With Prefect, you can build resilient, dynamic workflows that react to the world around them and recover from unexpected changes. With just a few decorators, Prefect supercharges your code with features like automatic retries, distributed execution, scheduling, caching, and much more. Every activity is tracked and observable with the Prefect server or Prefect Cloud dashboard.
+Prefect is an orchestrator for data-intensive workflows. Powered by an asynchronous orchestration engine, its the easiest way to transform any Python function into a unit of work that can be observed and governed. With Prefect, you can build resilient, dynamic workflows that react to the world around them and recover from unexpected changes. With just a few decorators, Prefect supercharges your code with features like automatic retries, distributed execution, scheduling, caching, and much more. Every activity is tracked and observable with the Prefect server or Prefect Cloud dashboard.
```
Comment (billpalombi) [inline README.md:23]:
```suggestion
Prefect is an orchestrator for data-intensive workflows. Its the easiest way to transform any Python function into a unit of work that can be observed and orchestrated. With Prefect, you can build resilient, dynamic workflows that react to the world around them and recover from unexpected changes. With just a few decorators, Prefect supercharges your code with features like automatic retries, distributed execution, scheduling, caching, and much more. Every activity is tracked and observable with the Prefect server or Prefect Cloud dashboard.
```

File: docs/tutorials/index.md [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+    - basics
+    - tasks
+    - flows
+    - subflows
+---
+# Tutorial Overview
+
+### Prerequisites
+
+Before you start, install Prefect in your Python virtual environment:
```
Comment (billpalombi) [inline docs/tutorials/index.md:16]:
```suggestion
Before you start, install Prefect:
```

**Review by billpalombi — COMMENTED** [review]
This is definitely a change for the better, especially with regards to style and structure. Thank you @rpeden and @discdiver!

File: docs/index.md [hunk 5]
```diff
@@ -1,224 +1,108 @@
...
+hide:
+  - toc
 ---
 
-# Welcome to Prefect 2
 
-!!! tip "Changing 'Orion' nomenclature"
-    With the 2.8.1 release, **we removed references to "Orion" and replaced them with more explicit, conventional nomenclature throughout the codebase**. These changes clarify the function of various components, commands, variables, and more. See the [Release Notes](https://github.com/PrefectHQ/prefect/blob/main/RELEASE-NOTES.md#release-281) for details.
+# **Welcome to Prefect**
+### The easiest way to orchestrate and observe your Python code { .main-page-subheader }
```
[the remark this answers is not in the document]
Comment (cicdw) [reply docs/index.md:15]: the phrase "observe your Python code" starts to sound too much like a tracer, and isn't really the unit of observation we're most focused on

File: docs/index.md [hunk 6]
```diff
@@ -1,224 +1,108 @@
...
-
-**Notifications**
-
-Easily set up e-mail, Slack, or PagerDuty notifications so that the right people are notified when something doesn't go as planned. 
-
-**Designed for performance**
+Prefect enables you to build resilient data workflows so that you can understand, react to, and recover from unexpected changes. Bring your Python code, sprinkle in a few decorators, and go!
 
-Prefect 2 has been designed from the ground up to handle the dynamic, scalable workloads that today's dataflows demands. 
+With Prefect you gain easy:
```
Comment (cicdw) [inline docs/index.md:19]:
```suggestion
With Prefect you gain simple:
```

File: docs/index.md [hunk 7]
```diff
@@ -1,224 +1,108 @@
...
-Fire up the Prefect UI locally by entering this command in your terminal:
+@flow
+def my_favorite_function():
+    print("What is your favorite number?")
+    return 42
+
+print(my_favorite_function())
+```
+
+Thats it! Your function is now a flow. Run the code as you normally would, and you'll see its orchestration log:
```
Comment (cicdw) [inline docs/index.md:74]:
```suggestion
Thats it! Your function is now a flow. Run the code as you normally would, and you'll see its execution via the Prefect logs:
```

File: docs/index.md [hunk 8]
```diff
@@ -1,224 +1,108 @@
...
+15:27:42.543 | INFO    | prefect.engine - Created flow run 'olive-poodle' for flow 'my-favorite-function'
+15:27:42.543 | INFO    | Flow run 'olive-poodle' - Using task runner 'ConcurrentTaskRunner'
+What is your favorite number?
+15:27:42.652 | INFO    | Flow run 'olive-poodle' - Finished in state Completed()
+42
 ```
 </div>
 
-Follow the link in your terminal to see the dashboard.
+The Prefect orchestration engine observes the flow run state and saves it in a database. By adding a single decorator, you've gained some observability of your flow. There's much more you can do with Prefect, so keep rolling to learn how to add easy retries, notifications, scheduling and more!
```
Comment (cicdw) [inline docs/index.md:87]:
```suggestion
Prefect automatically persists all executions of your flow along with useful metadata such as start time, end time, and state. This is just the beginning, so keep exploring to learn how to add retries, notifications, scheduling and much more!
```

File: docs/index.md [hunk 9]
```diff
@@ -1,224 +1,108 @@
...
 
-![screenshot of the Prefect dashboard with flow runs in a scatter plot](/img/intro-ui-dashboard.png)
+## Next Steps
+If you like what you see, try our [tutorials](/tutorials) and [guides](/guides), or read more about [Prefect concepts](/concepts).
```
Comment (cicdw) [inline docs/index.md:90]:
```suggestion
To learn more, try our [tutorials](/tutorials) and [guides](/guides), or go deeper with [Prefect concepts](/concepts).
```

File: docs/integrations/contribute.md [hunk 10]
```diff
@@ -59,23 +59,23 @@ class Secret(Block):
...
-Anyone can create and share a Prefect Collection and we encourage anyone interested in creating a collection to do so!
+Anyone can create and share a Prefect Integration and we encourage anyone interested in creating a integration to do so!
 
 ### Generate a project
 
-To help you get started with your collection, we've created a template that gives the tools you need to create and publish your collection.
+To help you get started with your integration, we've created a template that gives the tools you need to create and publish your integration.
 
-Use the [Prefect Collection template](https://github.com/PrefectHQ/prefect-collection-template#quickstart) to get started creating a collection with a bootstrapped project!
+Use the [Prefect Integration template](https://github.com/PrefectHQ/prefect-integration-template#quickstart) to get started creating a integration with a bootstrapped project!
```
Comment (cicdw) [inline docs/integrations/contribute.md:70]:
```suggestion
Use the [Prefect Integration template](https://github.com/PrefectHQ/prefect-integration-template#quickstart) to get started creating an integration with a bootstrapped project!
```

**Review by cicdw — CHANGES_REQUESTED** [review]
a few copy edits, thanks for putting in the effort to revamp all of this!

File: docs/index.md [hunk 11]
```diff
@@ -1,224 +1,109 @@
...
-
-Inevitably dataflows will fail. Prefect helps your code automatically retry on failure. 
-
-**Notifications**
-
-Easily set up e-mail, Slack, or PagerDuty notifications so that the right people are notified when something doesn't go as planned. 
+Prefect enables you to build resilient data workflows so that you can understand, react to, and recover from unexpected changes.
 
-**Designed for performance**
+Bring your Python code, sprinkle in a few decorators, and go!
```
[the remark this answers is not in the document]
Comment (discdiver) [reply docs/index.md]: incorporated by hand b/c outdated

File: docs/index.md [hunk 12]
```diff
@@ -1,224 +1,109 @@
...
-## Prefect highlights
-
-**Graceful failures**
-
-Inevitably dataflows will fail. Prefect helps your code automatically retry on failure. 
-
-**Notifications**
-
-Easily set up e-mail, Slack, or PagerDuty notifications so that the right people are notified when something doesn't go as planned. 
+Prefect enables you to build resilient data workflows so that you can understand, react to, and recover from unexpected changes.
```
[the remark this answers is not in the document]
Comment (discdiver) [reply docs/index.md]: incorporated manually

**Review by billpalombi — APPROVED** [review]
Thanks @discdiver!

**Review by cicdw — APPROVED** [review]

**Pull request merged**